packages/lib/src/components/ThreeDS2/callSubmit3DS2Fingerprint.ts

```diff
@@ -0,0 +1,59 @@
+import UIElement from '../UIElement';
+import fetchJSONData from '../../utils/fetch-json-data';
+
+/**
+ * ThreeDS2DeviceFingerprint, onComplete, calls a new, internal, endpoint which behaves like the /details endpoint but doesn't require the same credentials
+ */
+export default function callSubmit3DS2Fingerprint(state) {
+    fetchJSONData(
+        {
+            path: `v1/submitThreeDS2Fingerprint?token=${this.props.clientKey}`,
+            loadingContext: this.props.loadingContext,
+            method: 'POST',
+            contentType: 'application/json',
+            clientKey: this.props.clientKey
+        },
+        {
+            fingerprintResult: state.data.details[this.props.dataKey],
+            paymentData: state.data.paymentData
+        }
+    ).then(data => {
+        // elementRef exists when the fingerprint component is created from the Dropin
+        const actionHandler = this.props.elementRef ?? this;
+
+        if (!data.action) {
+            console.error('Handled Error::callSubmit3DS2Fingerprint::FAILED:: data=', data);
+            return;
+        }
+
+        /**
+         * Frictionless (no challenge) flow
+         */
+        if (data.action?.type === 'authenticationFinished') {
+            const detailsObj = {
+                data: {
+                    // Sending no details property should work - BUT currently doesn't in the new flow
+                    details: { 'threeds2.challengeResult': btoa('{"transStatus":"Y"}') },
+                    paymentData: data.action.paymentData
+                }
+            };
+
+            return UIElement.prototype.onComplete.call(this, detailsObj);
+        }
+
+        /**
+         * Challenge flow
+         */
+        if (data.action?.type === 'threeDS2') {
+            return actionHandler.handleAction(data.action);
+        }
+
+        /**
+         * Redirect (usecase: we thought we could do 3DS2 but it turns out we can't)
+         */
+        if (data.action?.type === 'redirect') {
+            data.action.paymentMethodType = 'scheme';
```

Comment:
Why do we need to override the `paymentMethodType`?

Reply:
Because the response for the `redirect` action is not returning a `paymentMethodType` - so the `handleAction` for "redirect" will fail

Reply:
It's not clear yet how easy it is for the endpoint to set this value in the response

Reply:
This has been changed in the endpoint - we no longer need to set paymentMethodType on "redirect" action